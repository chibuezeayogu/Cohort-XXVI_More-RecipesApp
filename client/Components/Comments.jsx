import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
