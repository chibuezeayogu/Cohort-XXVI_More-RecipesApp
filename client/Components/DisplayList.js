import React, { Component } from 'react';
import { Link } from 'react-router';

const Displaylist = React.createClass ({
    render(){
        const { recipes } = this.props;
        return(
            <td>
                list ingradients
            </td>
        );
    }
});

export default Displaylist;