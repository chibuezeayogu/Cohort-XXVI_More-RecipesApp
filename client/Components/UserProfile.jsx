import React, { Component } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { editProfile, fetchUser } from '../action/actionCreators';
import UserMenu from './Header/UserMenu.jsx';
import Footer from './Footer/Footer.jsx';
import SmallPreloader from './SmallPreloader.jsx';
import { validateUpdateProfileForm } from '../util/validateInputs';
import imageToFormData from '../util/ImageUpload';


/**
 * @description
 *
 * @class
 *
 * @extends Component
 *
 */
export class UserProfile extends Component {
  /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf UserProfile
   *
   * @returns {undefined}
   *
   */
  constructor() {
    super();
    this.state = {
      id: '',
      firstName: '',
      lastName: '',
      phone: '',
      imageUrl:'',
      image: {},
      location: '',
      address: '',
      errors: {},
      disabled: true,
      onUpdate: false
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
    this.handleImageChange = this.handleImageChange.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
    this.handleCancel = this.handleCancel.bind(this);
  }

  /**
   * @description dispatch an action to fetch user details
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @param {Object} nextProps
   *
   * @returns {undefiend}
   *
   */

  componentWillMount() {
    const { id } = this.props.userData.currentUser;
    this.props.fetchUser(id);
  }

  /**
   * @description sets State for input fileds
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @param {Object} nextProps
   *
   * @returns {undefined}
   *
   */
  componentWillReceiveProps(nextProps) {
    const { currentUser } = nextProps.userData;
    this.setState({
      disabled: true,
      id: currentUser.id,
      firstName: currentUser.firstName || '',
      lastName: currentUser.lastName || '',
      phone: currentUser.phone || '',
      location: currentUser.location|| '',
      address: currentUser.address || '',
      isLoading: false,
    });
  }

  /**
   * @description gets selected image
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @param {Object} event
   *
   * @returns {undefined}
   *
   */
  handleImageChange(event) {
    event.preventDefault();
    const file = event.target.files[0];
    this.setState({ errors: {} });

    this.setState({ errors: {}, image: file });
  }

  /**
   * @description gets user input
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @param {Object} event
   *
   * @returns {undefined}
   *
   */
  handleChange(event) {
    this.setState({ errors: {} });
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description sets input field to editable
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @param {Object} event
   *
   * @returns {undefined}
   *
   */
  handleEdit(event) {
    event.preventDefault();
    this.setState({ disabled: false });
  }

  /**
   * @description sets input field to readonly
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @param {Object} event
   *
   * @returns {undefined}
   *
   */
  handleCancel(event) {
    event.preventDefault();
    this.setState({ disabled: true });
  }

  /**
   * @description submits edit profile form
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @param {Object} event
   *
   * @returns {undefined}
   *
   */
  handleOnsubmit(event) {
    event.preventDefault();
    const err = validateUpdateProfileForm(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }
    this.setState({ disabled: true, onUpdate: true });
    if (this.state.image.name) {
      const uploadData = imageToFormData(this.state.image);
      delete axios.defaults.headers.common.Authorization;
      axios(uploadData)
        .then((data) => {
          this.setState({ imageUrl: data.data.secure_url });
          this.props.editProfile(this.state);
          this.setState({ onUpdate: false });
        }).catch(() => {
          this.setState({ disabled: false, onUpdate: false });
          Materialize.toast('Error! Please try again', 4000, 'red');
        });
    } else {
      this.props.editProfile(this.state);
      this.setState({ onUpdate: false });
    }
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf UserProfile
   *
   * @returns {undefined}
   */
  render() {
    const { imageUrl } = this.props.userData.currentUser;
    return (
      <div className="body">
        <UserMenu {...this.props} />
        <div className="main">
          <div className="container">
            <div className="row white" style={{ height: 100 }}>
              <h4 className="center">Profile </h4>
              <hr />
            </div>
            <form onSubmit={this.handleOnsubmit}>
              <div
                className="row profile-details-page"
              >
                <div className="col s12 m6 l5">
                  <img
                    src={imageUrl}
                    alt=""
                    className="responsive-img circle center profile-image"
                  />
                </div>
                <div className="col s12 m6 l7">
                  <div className="profile-input-display">
                    <div className="profile-label">FIRST NAME</div>
                    <input
                      id="firstName"
                      type="text"
                      name="firstName"
                      className="profile-input"
                      disabled={this.state.disabled}
                      value={this.state.firstName}
                      onChange={this.handleChange}
                    />
                    <span className="right red-text error-margin">
                      {this.state.errors.firstNameError}
                    </span>
                  </div>
                  <div className="profile-input-display">
                    <div className="profile-label">LAST NAME</div>
                    <input
                      type="text"
                      name="lastName"
                      className="profile-input"
                      disabled={this.state.disabled}
                      value={this.state.lastName}
                      onChange={this.handleChange}
                    />
                    <span className="right red-text error-margin">
                      {this.state.errors.lastNameError}
                    </span>
                  </div>
                  <div
                    className="profile-input-display"
                  >
                    <div
                      className="profile-label"
                    >PHONE
                    </div>
                    <input
                      type="text"
                      name="phone"
                      className="profile-input"
                      disabled={this.state.disabled}
                      value={this.state.phone}
                      onChange={this.handleChange}
                    />
                    <span className="right red-text error-margin">
                      {this.state.errors.phoneError}
                    </span>
                  </div>
                  <div className="profile-input-display">
                    <div className="profile-label">LOCATION</div>
                    <input
                      type="text"
                      name="location"
                      className="profile-input"
                      disabled={this.state.disabled}
                      value={this.state.location}
                      onChange={this.handleChange}
                    />
                    <span className="right red-text error-margin">
                      {this.state.errors.locationError}
                    </span>
                  </div>
                  <div
                    className="profile-input-display"
                  >
                    <div className="profile-label">ADDRESS</div>
                    <input
                      type="text"
                      name="address"
                      className="profile-input"
                      disabled={this.state.disabled}
                      value={this.state.address}
                      onChange={this.handleChange}
                    />
                    <span className="right red-text error-margin">
                      {this.state.errors.addressError}
                    </span>
                  </div>
                  <div
                    className="file-field input-field"
                    style={
                      this.state.disabled
                      ? { display: 'none' }
                      : { display: 'block' }
                    }
                  >
                    <div className="btn green">
                      <span>
                        <i
                          className="material-icons"
                        >insert_photo
                        </i>
                      </span>
                      <input
                        id="image"
                        type="file"
                        multiple
                        onChange={this.handleImageChange}
                      />
                    </div>
                    <div
                      className="file-path-wrapper"
                    >
                      <input
                        className="file-path validate"
                        type="text"
                        placeholder="Upload profile image"
                      />
                    </div>
                    <span className="right red-text error-margin">
                      {this.state.errors.imageError}
                    </span>
                  </div>
                  <div>
                    <button
                      id="edit"
                      className="btn modal-trigger green"
                      onClick={this.handleEdit}
                      style={{ marginRight: 5 }}
                    >Edit
                    </button>
                    <button
                      id="update"
                      className="btn green"
                      disabled={this.state.disabled}
                      style={{ marginRight: 5 }}
                    >Update
                    </button>
                    <button
                      id="cancel"
                      className="btn red lighten-1"
                      onClick={this.handleCancel}
                    >
                      Cancel
                    </button>
                  </div>
                  <hr />
                  <div className="row center">
                    <div>
                      {this.state.onUpdate ?
                        <SmallPreloader />
                      :
                      ''
                      }
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  userData: state.userData
});

export default connect(mapStateToProps,
  { editProfile, fetchUser })(UserProfile);
