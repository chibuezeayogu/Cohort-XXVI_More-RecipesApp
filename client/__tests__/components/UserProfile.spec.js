import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import mochData from '../__mock__/actionMockData';
import { UserProfile } from '../../Components/UserProfile';

const props = {
  userData: {
    isAuthenticated: true,
    currentUser: {
      id: 1,
      email: 'chibuezeayogu@hotmail.com'
    }
  },
  editProfile: jest.fn(),
  fetchUser: jest.fn()
};

const state = {
  disabled: false,
  id: '',
  firstName: '',
  lastName: '',
  phone: '',
  location: '',
  address: '',
  isLoading: false,
  errors: {},
};

const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'firstName',
    value: 'Nelson',
    files: [{}]
  }
};

describe('<UserProfile />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('componentWillReceiveProps', () => {
  it('should update component with new props', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    const componentWillReceivePropsSpy = jest
      .spyOn(wrapper.instance(), 'componentWillReceiveProps');
    wrapper.setProps({
      ...props,
      userData:
      {
        currentUser: mochData.userEdit
      }
    });
    wrapper.setState({ disabled: true });
    expect(componentWillReceivePropsSpy).toHaveBeenCalled();
  });
});

describe('handleEdit function', () => {
  it('should set input field to editable', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.find('#edit').simulate('click', event);
    expect(wrapper.instance().state.disabled).toEqual(false);
  });
});

describe('handleCancel function', () => {
  it('should set input field to readonly', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.find('#cancel').simulate('click', event);
    expect(wrapper.instance().state.disabled).toEqual(true);
  });
});

describe('handleImageChange function', () => {
  it('should set image state to user selected image', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.find('#image').simulate('change', event);
    expect(wrapper.instance().state.image).toEqual({});
  });
});

describe('handleChange function', () => {
  it('should add user details to state', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.find('#firstName').simulate('change', event);
    expect(wrapper.instance().state.firstName).toBe('Nelson');
  }
  );
});

describe('handleOnSubmit function', () => {
  it('should call handleChange function', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().state.errors.firstNameError)
      .toBe('first name is required');
  });

  it('should call handleChange function', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().state.errors.firstNameError)
      .toBe('first name is required');
  });
});

describe('HandleOnSubmit function', () => {
  it('should return error if input fields are empty', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().state.errors.firstNameError)
      .toBe('first name is required');
    expect(wrapper.instance().state.errors.lastNameError)
      .toBe('last name is required');
    expect(wrapper.instance().state.errors.locationError)
      .toBe('location is required');
    expect(wrapper.instance().state.errors.addressError)
      .toBe('address is required');
    expect(wrapper.instance().state.errors.phoneError)
      .toBe('phone number is required');
  });

  it('should return error for input did not meet the required lenght', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.setState({
      firstName: 'ch',
      lastName: 'Ne',
      location: 'la',
      address: 'No',
      phone: 'pppp',
    });
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().state.errors.firstNameError)
      .toBe('first name must be at least 3 characters long');
    expect(wrapper.instance().state.errors.lastNameError)
      .toBe('last name must be at least 3 characters long');
    expect(wrapper.instance().state.errors.locationError)
      .toBe('location must be at least 5 characters long');
    expect(wrapper.instance().state.errors.addressError)
      .toBe('address name must be at least 8 characters long');
    expect(wrapper.instance().state.errors.phoneError)
      .toBe('phone value must be numbers');
  });

  it('should sumbit form if all conditions are meet', () => {
    const wrapper = shallow(<UserProfile {...props} {...state} />);
    wrapper.setState({
      firstName: 'chibueze',
      lastName: 'Nelson',
      location: 'lagos, Nigeria',
      address: 'No 1 Aminu street Mende',
      phone: '07033297338',
      image: { name: 'image', type: 'image/png' },
    });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(wrapper.instance().state.disabled).toEqual(true);
  });
});
