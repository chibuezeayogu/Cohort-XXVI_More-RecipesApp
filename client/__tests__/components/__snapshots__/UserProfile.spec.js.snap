// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body"
  >
    <withRouter(Connect(UserMenu))
      editProfile={[MockFunction]}
      fetchUser={
        [MockFunction] {
          "calls": Array [
            Array [
              1,
            ],
          ],
        }
      }
      userData={
        Object {
          "currentUser": Object {
            "email": "chibuezeayogu@hotmail.com",
            "id": 1,
          },
          "isAuthenticated": true,
        }
      }
    />
    <div
      className="main"
    >
      <div
        className="container"
      >
        <div
          className="row white"
          style={
            Object {
              "height": 100,
            }
          }
        >
          <h4
            className="center"
          >
            Profile 
          </h4>
          <hr />
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="row profile-details-page"
          >
            <div
              className="col s12 m6 l5"
            >
              <img
                alt=""
                className="responsive-img circle center profile-image"
                src={undefined}
              />
            </div>
            <div
              className="col s12 m6 l7"
            >
              <div
                className="profile-input-display"
              >
                <div
                  className="profile-label"
                >
                  FIRST NAME
                </div>
                <input
                  className="profile-input"
                  disabled={true}
                  id="firstName"
                  name="firstName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="profile-input-display"
              >
                <div
                  className="profile-label"
                >
                  LAST NAME
                </div>
                <input
                  className="profile-input"
                  disabled={true}
                  name="lastName"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="profile-input-display"
              >
                <div
                  className="profile-label"
                >
                  PHONE
                </div>
                <input
                  className="profile-input"
                  disabled={true}
                  name="phone"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="profile-input-display"
              >
                <div
                  className="profile-label"
                >
                  LOCATION
                </div>
                <input
                  className="profile-input"
                  disabled={true}
                  name="location"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="profile-input-display"
              >
                <div
                  className="profile-label"
                >
                  ADDRESS
                </div>
                <input
                  className="profile-input"
                  disabled={true}
                  name="address"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="file-field input-field"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <div
                  className="btn green"
                >
                  <span>
                    <i
                      className="material-icons"
                    >
                      insert_photo
                    </i>
                  </span>
                  <input
                    id="image"
                    multiple={true}
                    onChange={[Function]}
                    type="file"
                  />
                </div>
                <div
                  className="file-path-wrapper"
                >
                  <input
                    className="file-path validate"
                    placeholder="Upload profile image"
                    type="text"
                  />
                </div>
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div>
                <button
                  className="btn modal-trigger green"
                  id="edit"
                  onClick={[Function]}
                  style={
                    Object {
                      "marginRight": 5,
                    }
                  }
                >
                  Edit
                </button>
                <button
                  className="btn green"
                  disabled={true}
                  id="update"
                  style={
                    Object {
                      "marginRight": 5,
                    }
                  }
                >
                  Update
                </button>
                <button
                  className="btn red lighten-1"
                  id="cancel"
                  onClick={[Function]}
                >
                  Cancel
                </button>
              </div>
              <hr />
              <div
                className="row center"
              >
                <div>
                  
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body"
    >
      <withRouter(Connect(UserMenu))
        editProfile={[MockFunction]}
        fetchUser={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          }
        }
      />
      <div
        className="main"
      >
        <div
          className="container"
        >
          <div
            className="row white"
            style={
              Object {
                "height": 100,
              }
            }
          >
            <h4
              className="center"
            >
              Profile 
            </h4>
            <hr />
          </div>
          <form
            onSubmit={[Function]}
          >
            <div
              className="row profile-details-page"
            >
              <div
                className="col s12 m6 l5"
              >
                <img
                  alt=""
                  className="responsive-img circle center profile-image"
                  src={undefined}
                />
              </div>
              <div
                className="col s12 m6 l7"
              >
                <div
                  className="profile-input-display"
                >
                  <div
                    className="profile-label"
                  >
                    FIRST NAME
                  </div>
                  <input
                    className="profile-input"
                    disabled={true}
                    id="firstName"
                    name="firstName"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <span
                    className="right red-text error-margin"
                  />
                </div>
                <div
                  className="profile-input-display"
                >
                  <div
                    className="profile-label"
                  >
                    LAST NAME
                  </div>
                  <input
                    className="profile-input"
                    disabled={true}
                    name="lastName"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <span
                    className="right red-text error-margin"
                  />
                </div>
                <div
                  className="profile-input-display"
                >
                  <div
                    className="profile-label"
                  >
                    PHONE
                  </div>
                  <input
                    className="profile-input"
                    disabled={true}
                    name="phone"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <span
                    className="right red-text error-margin"
                  />
                </div>
                <div
                  className="profile-input-display"
                >
                  <div
                    className="profile-label"
                  >
                    LOCATION
                  </div>
                  <input
                    className="profile-input"
                    disabled={true}
                    name="location"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <span
                    className="right red-text error-margin"
                  />
                </div>
                <div
                  className="profile-input-display"
                >
                  <div
                    className="profile-label"
                  >
                    ADDRESS
                  </div>
                  <input
                    className="profile-input"
                    disabled={true}
                    name="address"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <span
                    className="right red-text error-margin"
                  />
                </div>
                <div
                  className="file-field input-field"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  <div
                    className="btn green"
                  >
                    <span>
                      <i
                        className="material-icons"
                      >
                        insert_photo
                      </i>
                    </span>
                    <input
                      id="image"
                      multiple={true}
                      onChange={[Function]}
                      type="file"
                    />
                  </div>
                  <div
                    className="file-path-wrapper"
                  >
                    <input
                      className="file-path validate"
                      placeholder="Upload profile image"
                      type="text"
                    />
                  </div>
                  <span
                    className="right red-text error-margin"
                  />
                </div>
                <div>
                  <button
                    className="btn modal-trigger green"
                    id="edit"
                    onClick={[Function]}
                    style={
                      Object {
                        "marginRight": 5,
                      }
                    }
                  >
                    Edit
                  </button>
                  <button
                    className="btn green"
                    disabled={true}
                    id="update"
                    style={
                      Object {
                        "marginRight": 5,
                      }
                    }
                  >
                    Update
                  </button>
                  <button
                    className="btn red lighten-1"
                    id="cancel"
                    onClick={[Function]}
                  >
                    Cancel
                  </button>
                </div>
                <hr />
                <div
                  className="row center"
                >
                  <div>
                    
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserProfile
        editProfile={[MockFunction]}
        fetchUser={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        userData={
          Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserProfile {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleCancel": [Function],
        "handleChange": [Function],
        "handleEdit": [Function],
        "handleImageChange": [Function],
        "handleOnsubmit": [Function],
        "props": Object {
          "editProfile": [MockFunction],
          "fetchUser": [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          },
          "userData": Object {
            "currentUser": Object {
              "email": "chibuezeayogu@hotmail.com",
              "id": 1,
            },
            "isAuthenticated": true,
          },
        },
        "refs": Object {},
        "state": Object {
          "address": "",
          "disabled": true,
          "errors": Object {},
          "firstName": "",
          "id": "",
          "image": Object {},
          "imageUrl": "",
          "lastName": "",
          "location": "",
          "onUpdate": false,
          "phone": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body"
        >
          <withRouter(Connect(UserMenu))
            editProfile={[MockFunction]}
            fetchUser={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            userData={
              Object {
                "currentUser": Object {
                  "email": "chibuezeayogu@hotmail.com",
                  "id": 1,
                },
                "isAuthenticated": true,
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="row white"
                style={
                  Object {
                    "height": 100,
                  }
                }
              >
                <h4
                  className="center"
                >
                  Profile 
                </h4>
                <hr />
              </div>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="row profile-details-page"
                >
                  <div
                    className="col s12 m6 l5"
                  >
                    <img
                      alt=""
                      className="responsive-img circle center profile-image"
                      src={undefined}
                    />
                  </div>
                  <div
                    className="col s12 m6 l7"
                  >
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        FIRST NAME
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        id="firstName"
                        name="firstName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        LAST NAME
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="lastName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        PHONE
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="phone"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        LOCATION
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="location"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        ADDRESS
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="address"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="file-field input-field"
                      style={
                        Object {
                          "display": "none",
                        }
                      }
                    >
                      <div
                        className="btn green"
                      >
                        <span>
                          <i
                            className="material-icons"
                          >
                            insert_photo
                          </i>
                        </span>
                        <input
                          id="image"
                          multiple={true}
                          onChange={[Function]}
                          type="file"
                        />
                      </div>
                      <div
                        className="file-path-wrapper"
                      >
                        <input
                          className="file-path validate"
                          placeholder="Upload profile image"
                          type="text"
                        />
                      </div>
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div>
                      <button
                        className="btn modal-trigger green"
                        id="edit"
                        onClick={[Function]}
                        style={
                          Object {
                            "marginRight": 5,
                          }
                        }
                      >
                        Edit
                      </button>
                      <button
                        className="btn green"
                        disabled={true}
                        id="update"
                        style={
                          Object {
                            "marginRight": 5,
                          }
                        }
                      >
                        Update
                      </button>
                      <button
                        className="btn red lighten-1"
                        id="cancel"
                        onClick={[Function]}
                      >
                        Cancel
                      </button>
                    </div>
                    <hr />
                    <div
                      className="row center"
                    >
                      <div>
                        
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body"
        >
          <withRouter(Connect(UserMenu))
            editProfile={[MockFunction]}
            fetchUser={
              [MockFunction] {
                "calls": Array [
                  Array [
                    1,
                  ],
                ],
              }
            }
            userData={
              Object {
                "currentUser": Object {
                  "email": "chibuezeayogu@hotmail.com",
                  "id": 1,
                },
                "isAuthenticated": true,
              }
            }
          />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="row white"
                style={
                  Object {
                    "height": 100,
                  }
                }
              >
                <h4
                  className="center"
                >
                  Profile 
                </h4>
                <hr />
              </div>
              <form
                onSubmit={[Function]}
              >
                <div
                  className="row profile-details-page"
                >
                  <div
                    className="col s12 m6 l5"
                  >
                    <img
                      alt=""
                      className="responsive-img circle center profile-image"
                      src={undefined}
                    />
                  </div>
                  <div
                    className="col s12 m6 l7"
                  >
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        FIRST NAME
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        id="firstName"
                        name="firstName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        LAST NAME
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="lastName"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        PHONE
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="phone"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        LOCATION
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="location"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="profile-input-display"
                    >
                      <div
                        className="profile-label"
                      >
                        ADDRESS
                      </div>
                      <input
                        className="profile-input"
                        disabled={true}
                        name="address"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div
                      className="file-field input-field"
                      style={
                        Object {
                          "display": "none",
                        }
                      }
                    >
                      <div
                        className="btn green"
                      >
                        <span>
                          <i
                            className="material-icons"
                          >
                            insert_photo
                          </i>
                        </span>
                        <input
                          id="image"
                          multiple={true}
                          onChange={[Function]}
                          type="file"
                        />
                      </div>
                      <div
                        className="file-path-wrapper"
                      >
                        <input
                          className="file-path validate"
                          placeholder="Upload profile image"
                          type="text"
                        />
                      </div>
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                    <div>
                      <button
                        className="btn modal-trigger green"
                        id="edit"
                        onClick={[Function]}
                        style={
                          Object {
                            "marginRight": 5,
                          }
                        }
                      >
                        Edit
                      </button>
                      <button
                        className="btn green"
                        disabled={true}
                        id="update"
                        style={
                          Object {
                            "marginRight": 5,
                          }
                        }
                      >
                        Update
                      </button>
                      <button
                        className="btn red lighten-1"
                        id="cancel"
                        onClick={[Function]}
                      >
                        Cancel
                      </button>
                    </div>
                    <hr />
                    <div
                      className="row center"
                    >
                      <div>
                        
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserProfile
    editProfile={[MockFunction]}
    fetchUser={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
      }
    }
    userData={
      Object {
        "currentUser": Object {
          "email": "chibuezeayogu@hotmail.com",
          "id": 1,
        },
        "isAuthenticated": true,
      }
    }
  />,
}
`;
